homeassistant/components/iron_os/entity.py

```diff
@@ -31,11 +32,21 @@ def __init__(
         )
         if TYPE_CHECKING:
             assert coordinator.config_entry.unique_id
-        self.device_info = DeviceInfo(
+
+        serial_number = None
+        if coordinator.config_entry.data.get(CONF_SERIAL_NUMBER):
+            serial_number = f"{coordinator.config_entry.data.get(CONF_SERIAL_NUMBER)} DeviceID:{coordinator.config_entry.data.get(CONF_DEVICE_ID)}"
+
+        self._attr_device_info = DeviceInfo(
             connections={(CONNECTION_BLUETOOTH, coordinator.config_entry.unique_id)},
             manufacturer=MANUFACTURER,
             model=MODEL,
             name="Pinecil",
-            sw_version=coordinator.device_info.build,
-            serial_number=f"{coordinator.device_info.device_sn} (ID:{coordinator.device_info.device_id})",
+            sw_version=coordinator.config_entry.data.get(CONF_SW_VERSION),
+            serial_number=serial_number,
         )
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return super().available and self.coordinator.device._client.is_connected  # noqa: SLF001
```

Comment:
why the noqa?

Reply:
accessing a private member of bleak

Reply:
oh I see, why do we do that?

Reply:
The device went unavailable when the coordinator update failed, but as it doesn't raise `UpdateFailed` anymore, I set the available status based on if the device is connected or not

Reply:
No I mean, why is the thing private in the library, if we want to use it, shouldn't it be public?

Reply:
Or become, I think I am also fine with fixing this in a followup

Reply:
Had planned to do that anyway. If you are ok to do a lib bump in this PR, I can change it right away, otherwise I'll fix this in a separate lib bump PR